test/options_test.dart

```diff
@@ -584,4 +584,50 @@ class Foo {}
         throwsA(const TypeMatcher<DartdocFailure>().having((f) => f.message,
             'message', startsWith('Missing required template file'))));
   });
+
+  group('limit files created', () {
+    test('maxFileCount', () async {
```

Comment:
Let's add a test for each case where a Failure is not thrown, one for maxFileCount and one for maxTotalSize